src/compose/compose-scalar.ts

```diff
@@ -53,6 +53,12 @@ export function composeScalar(
   if (tagName) scalar.tag = tagName
   if (tag.format) scalar.format = tag.format
   if (comment) scalar.comment = comment
+  if (ctx.options.keepSourceTokens && token.type === 'block-scalar') {
+    const headerToken = token.props[0]
```

Comment:
The access to `token.props[0]` is guaranteed to be safe by the parser structure and is consistent with existing code patterns in this repository.

When a `BlockScalar` token is created, it's [always initialized](https://github.com/eemeli/yaml/blob/92821f2b8164f9831ff5a51f6e5a575e06365742/src/parse/parser.ts#L870-L876) with the `block-scalar-header` token as the first element in `props`.

The codebase already relies on this guarantee in the same way at [`cst-scalar.ts:264`](https://github.com/eemeli/yaml/blob/92821f2b8164f9831ff5a51f6e5a575e06365742/src/parse/cst-scalar.ts#L264).